lib/taskana-core/src/main/java/pro/taskana/spi/priority/internal/PriorityServiceManager.java

```diff
@@ -0,0 +1,88 @@
+package pro.taskana.spi.priority.internal;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pro.taskana.common.api.exceptions.SystemException;
+import pro.taskana.common.internal.util.LogSanitizer;
+import pro.taskana.spi.priority.api.PriorityServiceProvider;
+import pro.taskana.task.api.models.TaskSummary;
+
+public class PriorityServiceManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PriorityServiceManager.class);
+  private static PriorityServiceManager singleton;
+  private final ServiceLoader<PriorityServiceProvider> serviceLoader;
+  private boolean enabled = false;
+
+  private PriorityServiceManager() {
+    serviceLoader = ServiceLoader.load(PriorityServiceProvider.class);
+    for (PriorityServiceProvider priorityProvider : serviceLoader) {
+      LOGGER.info("Registered PriorityServiceProvider: {}", priorityProvider.getClass().getName());
+      enabled = true;
+    }
+    if (!enabled) {
+      LOGGER.info("No PriorityServiceProvider found. Running without PriorityServiceProvider.");
+    }
+  }
+
+  public static synchronized PriorityServiceManager getInstance() {
+    if (singleton == null) {
+      singleton = new PriorityServiceManager();
+    }
+    return singleton;
+  }
+
+  public static boolean isPriorityServiceEnabled() {
+    return Objects.nonNull(singleton) && singleton.enabled;
+  }
+
+  public Optional<Integer> calculatePriorityOfTask(TaskSummary task) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Sending task to PriorityServiceProviders: {}", task);
+    }
+
+    Set<Integer> priorities = new HashSet<>();
+    for (PriorityServiceProvider provider : serviceLoader) {
+      Optional<Integer> priority = getPriorityByProvider(task, provider);
+      if (priority.isPresent()) {
+        priorities.add(priority.get());
+      }
+    }
```

Comment:
I know that you've commented above that working with a stream does not work. Can you explain why?
The serviceLoader provides a stream function. If that doesn't work I'd personally not use the serviceProvider variable within the entity. But you don't have to do that. We'll optimize that and the comment below with the local SPI's for testing ticket. This is just letting you know :)
Anyway: It's a bad practice to use the `Optional#isPresent`, because it's an indicator that you're not working pure functional.
Better: `getPriorityByProvider(task,provider).ifPresent(priorities::add)`

Reply:
We tried the following:
```
List<Integer> priorities =
        serviceLoader.stream()
            .map(provider -> getPriorityByProvider(task, provider))
            .filter(Optional::isPresent)
            .map(Optional::get)
            .distinct()
            .collect(Collectors.toList());
```
Then we have to include ServiceLoader.Provider which strangely won't get compiled, because it does not find the specified import.